app/components/documents/draft-document-card.js

```diff
@@ -82,6 +83,12 @@ export default class DocumentsDraftDocumentCardComponent extends Component {
     return isPresent(this.args.bordered) ? this.args.bordered : true;
   }
 
+  get mayDelete() {
+    return (
+      this.retrievedPieces?.length === 0 || this.currentSession.may('remove-piece-from-parliament')
```

Comment:
Draft pieces can never have retrievedPieces, right?

So this will always be falsy for secretarie and minister/KDB I think.

Also, currentSession is not defined here...

Was this tested?

Reply:
removed the code from draft-document-card.
The `model.pieces` here should only be draft-pieces which don't have the `retrieved-piece` relation.
Also, it was now impossible to remove any draft piece as non admin even if it was your own submission.